exporter/loadbalancingexporter/factory.go

```diff
@@ -32,20 +40,94 @@ func createDefaultConfig() component.Config {
 	otlpDefaultCfg.Endpoint = "placeholder:4317"
 
 	return &Config{
+		TimeoutSettings: exporterhelper.NewDefaultTimeoutConfig(),
```

Comment:
This behavior should be clearly documented. I think it would even make sense to log a warning in case users tried to use that option.

That said, there ARE users relying on this feature at the moment. What should we do about them? I think we should copy their current values to the load-balancer level, which would give them roughly the same desired behavior they have today.

So, here's how it could work:

1) if the OTLP Exporter options do not include the resiliency options, use our new defaults 
2) if there are, copy the ones from the OTLP section into the top-level config, and set the OTLP one to the default values, writing a log message stating that this is happening
3) if there are options at both levels, the ones at the load-balancing level wins, as that's the newest one and we can assume it's the latest intent from the user (but log a WARN in this case, it's serious enough)

I don't think we need a feature flag or deprecation plan for this.

Reply:
Should we do this approach for all 3 resilience options (queue, retry and timeout) or it should be only to timeout settings?

Reply:
@jpkrohling re-ping - I think this is the only open question left on this PR. Trying to get it moving again

Reply:
> Should we do this approach for all 3 resilience options

Please do all three in this PR since it can go out together for the best optimal experience.

Reply:
I was thinking a lot about this "magic" and I'm sure now that we shouldn't do such settings copy because it may produce bunch of unpredictable issues.
First of all users not always looking into the logs, especially on containerized environments like K8s, and may easily miss warning record.
Also, to make this change work as expected, user needs to tune resilience options for BOTH `loadbalancing` exporter and `otlp` sub-exportes.
From our own experience, when we were testing this change: we have to lower down all 3 resilience options configured initially for sub-exportes to make data delivery reliable in our quite flexible environment. Simple copy of those settings make everything even worse than it was before
For example, retry and timeout settings were set to non-default, quite large, values. Keeping them as-is has a side effect of large delay in endpoints re-balancing, because failed sub-exporter was trying again and again to send data to already dead endpoint instead of returning data to `loadbalancing` exporter and re-balance it to "fresh" endpoints.
So I believe that users SHOULD review `loadbalancing` exporter settings by themself to achieve proper results instead of relying on unexpected change made by exporter code.
Probably we should add some additional warning to Changelog item that users should review their settings when upgrading to a new version? For me it sounds as much better option than implementing "magic" copy-paste code

Reply:
I think that makes sense, and given you've tested in production that gives me more confidence.

Reply:
Thank you for your response @Fiery-Fenix.

What I would suggest is the log a warning if the options are ignored at the load balancer level. Perhaps some internal metrics to track and some configuration to update based on the data being returned.

To @jpkrohling point, adding the load balancer queue and timeout helpers could cause some issues, mostly the timeout of timeouts for unaware users, so a log would help here. (Providing users more context and warnings, even if they may not get reviewed is important).

I do agree with your point that copying values might be magic and perhaps not the correct path forward. From my perspective, I think the top level load balancer would be at least double the sub exporters settings by default; mostly to allow a reshuffled into the queue and delivered to the next available endpoint.

The definite warning is when the top level exporter has a lower configuration compared to the sub exporter, that is a definite must.

Reply:
I guess I'm just confused, but I'll try to clarify my points:

- If this PR brings change in behavior for people using the load-balancing exporter today, this is potentially a breaking change. Breaking changes have to be phased in: at N+1 (next version), this new feature has to be added behind a feature-flag and disabled by default. with proper notices to end-users about the upcoming change in the default behavior (changelog, warn log entries); at N+3, the flag is enabled by default; at N+5, the flag is removed and only the new behavior is active. 
- My previous understanding was that the resiliency options from the individual exporters would be moved to the top-level config. Copying the config from the individual exporters config to the top-level would make this a non-breaking change, as the behavior for the end-user is the roughly the same.
- My understanding now is that this is not changing the behavior: people can still configure the resiliency options at the individual exporters *and* at the top-level exporter. In this case, the readme still seems to be wrong in one place (the sending queue comment I left in place).